src/coinjoin/client.cpp

```diff
@@ -1448,12 +1448,13 @@ bool CCoinJoinClientSession::MakeCollateralAmounts()
     });
 
     // First try to use only non-denominated funds
-    if (std::any_of(vecTally.begin(), vecTally.end(), [&](const auto& item) { return MakeCollateralAmounts(item, false); })) {
-        return true;
+    for (const auto& item : vecTally) {
+        if (MakeCollateralAmounts(item, false)) return true;
     }
+
     // There should be at least some denominated funds we should be able to break in pieces to continue mixing
-    if (std::any_of(vecTally.begin(), vecTally.end(), [&](const auto& item) { return MakeCollateralAmounts(item, true); })) {
-        return true;
+    for (const auto& item : vecTally) {
+        if (MakeCollateralAmounts(item, true)) return true;
```

Comment:
9619f1a708a387aec19864b108d2a7b5e3d897dc: looks unrelated

Reply:
It looks like unrelated, but actually related because std:: algorithms doesn't pass annotations and it causes compilation errors

Reply:
could use ranges here
```
    // First try to use only non-denominated funds
    if (ranges::any_of(vecTally, [&](const auto& item) EXCLUSIVE_LOCKS_REQUIRED(m_wallet.cs_wallet) {
            return MakeCollateralAmounts(item, false);
        })) {
        return true;
    }
    // There should be at least some denominated funds we should be able to break in pieces to continue mixing
    if (ranges::any_of(vecTally, [&](const auto& item) EXCLUSIVE_LOCKS_REQUIRED(m_wallet.cs_wallet) {
            return MakeCollateralAmounts(item, true);
        })) {
        return true;
    }
```